src/vaadin-split-layout.js

```diff
@@ -0,0 +1,360 @@
+        value: 'horizontal'
+      },
+
+      /** @private */
+      _previousPrimaryPointerEvents: String,
+
+      /** @private */
+      _previousSecondaryPointerEvents: String
+    };
+  }
+
+  /** @protected */
+  ready() {
+    super.ready();
+    new FlattenedNodesObserver(this, this._processChildren);
+  }
+
+  /** @private */
+  _processChildren() {
+    this.getEffectiveChildren().forEach((child, i) => {
+      if (i === 0) {
+        this._primaryChild = child;
+        child.setAttribute('slot', 'primary');
+      } else if (i == 1) {
+        this._secondaryChild = child;
+        child.setAttribute('slot', 'secondary');
+      } else {
+        child.removeAttribute('slot');
+      }
+    });
+  }
+
+  /** @private */
+  _setFlexBasis(element, flexBasis, containerSize) {
+    flexBasis = Math.max(0, Math.min(flexBasis, containerSize));
+    if (flexBasis === 0) {
+      // Pure zero does not play well in Safari
+      flexBasis = 0.000001;
+    }
+    element.style.flex = '1 1 ' + flexBasis + 'px';
+  }
+
+  /** @private */
+  _setPointerEventsNone(event) {
+    if (!this._primaryChild || !this._secondaryChild) {
+      return;
+    }
+    this._previousPrimaryPointerEvents = this._primaryChild.style.pointerEvents;
+    this._previousSecondaryPointerEvents = this._secondaryChild.style.pointerEvents;
+    this._primaryChild.style.pointerEvents = 'none';
+    this._secondaryChild.style.pointerEvents = 'none';
+
+    event.preventDefault();
+  }
+
+  /** @private */
+  _restorePointerEvents() {
+    if (!this._primaryChild || !this._secondaryChild) {
+      return;
+    }
+    this._primaryChild.style.pointerEvents = this._previousPrimaryPointerEvents;
+    this._secondaryChild.style.pointerEvents = this._previousSecondaryPointerEvents;
+  }
+
+  /** @private */
+  _onHandleTrack(event) {
+    if (!this._primaryChild || !this._secondaryChild) {
+      return;
+    }
+
+    var size = this.orientation === 'vertical' ? 'height' : 'width';
+    if (event.detail.state === 'start') {
+      this._startSize = {
+        container: this.getBoundingClientRect()[size] - this.$.splitter.getBoundingClientRect()[size],
+        primary: this._primaryChild.getBoundingClientRect()[size],
+        secondary: this._secondaryChild.getBoundingClientRect()[size]
+      };
+
+      return;
+    }
+
+    var distance = this.orientation === 'vertical' ? event.detail.dy : event.detail.dx;
+    const isRtl = this.orientation !== 'vertical' && this.getAttribute('dir') === 'rtl';
+    const dirDistance = isRtl ? -distance : distance;
+
+    this._setFlexBasis(this._primaryChild, this._startSize.primary + dirDistance, this._startSize.container);
+    this._setFlexBasis(this._secondaryChild, this._startSize.secondary - dirDistance, this._startSize.container);
+
+    this.notifyResize();
+
+    if (event.detail.state === 'end') {
+      this.dispatchEvent(new CustomEvent('splitter-dragend'));
+
+      delete this._startSize;
+    }
+  }
+
```

Comment:
Should we still have the explicit `notifyResize` here somewhere...? Now it gets removed from the API docs. Even if we dropped Polymer, we might still want to have the API available?

Reply:
Restored the method to show it in the API docs.